### PR TITLE
TF2 porting: set feature

### PR DESCRIPTION
**Pull request by ydudin3**
# Code Pull Requests

Please provide the following:

- a clear explanation of what your code dose
- if applicable, a reference to an issue
- a reproducible test for your PR (code, model definition and data sample)

# Documentation Pull Requests

Note that the documentation HTML files are in `docs/` while the Markdown sources are in `mkdocs/docs`.

If you are proposing a modification to the documentation you should change only the Markdown files.

`api.md` is automatically generated from the docstrings in the code, so if you want to change something in that file, first modify `ludwig/api.py` docstring, then run `mkdocs/code_docs_autogen.py`, which will create `mkdocs/docs/api.md` .

### CODE REVIEW

File: ludwig/features/set_feature.py [hunk 1]
```diff
@@ -78,13 +79,12 @@ def feature_data(column, metadata, preprocessing_parameters):
...
             (len(column),
              len(metadata['str2idx'])),
-            dtype=bool
         )
 
         for i in range(len(column)):
             set_matrix[i, feature_vector[i]] = 1
 
-        return set_matrix
+        return set_matrix.astype(np.int32)
```
Comment (w4nderlust) [inline ludwig/features/set_feature.py:87]: why int32? if I remember correctly this matrix was boolean

File: ludwig/features/set_feature.py [hunk 2]
```diff
@@ -102,68 +102,28 @@ def add_feature_data(
...
-        self.pretrained_embeddings = None
-        self.embeddings_on_cpu = False
-        self.dropout = False
-        self.initializer = None
-        self.regularize = True
+    encoder = 'sparse'
 
-        _ = self.overwrite_defaults(feature)
+    def __init__(self, feature, encoder_obj=None):
+        super().__init__(feature)
```
Comment (w4nderlust) [inline ludwig/features/set_feature.py:108]: i guess yo udon't need it anymore as you have the explicit call to the two superclasses

File: ludwig/models/modules/set_encoders.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            pretrained_embeddings=None,
+            embeddings_on_cpu=False,
+            dropout_rate=0.0,
+            initializer=None,
+            regularizer=None,
+            **kwargs
+    ):
+        super(SetSparseEncoder, self).__init__()
+
+        self.embed = Embed(
```
Comment (w4nderlust) [inline ludwig/models/modules/set_encoders.py:42]: I think yo ucan use EmbedSparse here, which accepts binary multiple hot matrices rather than int32s

File: ludwig/models/modules/loss_modules.py [hunk 4]
```diff
@@ -478,6 +494,29 @@ def weighted_softmax_cross_entropy(
...
+def sigmoid_cross_entropy(
+        logits,
+        targets,
+        class_weights=1,
+        labels_smoothing=0,
+        **kwargs
+):
+    use_class_weights = not isinstance(class_weights, (int, float))
+    if use_class_weights:
+        train_loss = softmax_cross_entropy_with_class_weighting(
```
Comment (w4nderlust) [inline ludwig/models/modules/loss_modules.py:506]: this is softmax not signoid

File: ludwig/models/modules/loss_modules.py [hunk 5]
```diff
@@ -478,6 +494,29 @@ def weighted_softmax_cross_entropy(
...
+            logits,
+            vector_labels,
+            class_weights,
+            labels_smoothing
+        )
+    else:
+        train_loss = tf.losses.softmax_cross_entropy(
```
Comment (w4nderlust) [inline ludwig/models/modules/loss_modules.py:513]: this is softmax not signoid

Comment (ydudin3) [reply ludwig/models/modules/loss_modules.py:513]: oops, let me fix that

File: ludwig/models/modules/loss_modules.py (hunk 4 above)
Comment (ydudin3) [reply ludwig/models/modules/loss_modules.py:506]: don't actually need this helper, let me clean up
